File: android-app/src/main/java/com/prevengos/plug/android/data/repository/CuestionarioRepository.java
package com.prevengos.plug.android.data.repository;

import androidx.lifecycle.LiveData;

import com.prevengos.plug.android.data.local.dao.CuestionarioDao;
import com.prevengos.plug.android.data.local.entity.CuestionarioEntity;
import com.prevengos.plug.android.data.local.entity.RespuestaLocal;

import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class CuestionarioRepository {
    private final CuestionarioDao cuestionarioDao;

    public CuestionarioRepository(CuestionarioDao cuestionarioDao) {
        this.cuestionarioDao = cuestionarioDao;
    }

    public LiveData<List<CuestionarioEntity>> observeForPaciente(String pacienteId) {
        return cuestionarioDao.observeByPaciente(pacienteId);
    }

    public CuestionarioEntity findById(String cuestionarioId) {
        return cuestionarioDao.findById(cuestionarioId);
    }

    public CuestionarioEntity createCuestionario(String pacienteId,
                                                 String plantillaCodigo,
                                                 String estado,
                                                 List<RespuestaLocal> respuestas) {
        long now = System.currentTimeMillis();
        CuestionarioEntity entity = new CuestionarioEntity(
                UUID.randomUUID().toString(),
                pacienteId,
                plantillaCodigo,
                estado,
                respuestas,
                Collections.emptyList(),
                Collections.emptyList(),
                null,
                null,
                now,
                null,
                true);
        cuestionarioDao.upsert(entity);
        return entity;
    }

    public CuestionarioEntity updateCuestionario(String cuestionarioId,
                                                 String estado,
                                                 List<RespuestaLocal> respuestas) {
        CuestionarioEntity existing = cuestionarioDao.findById(cuestionarioId);
        if (existing == null) {
            throw new IllegalArgumentException("Cuestionario no encontrado");
        }
        long now = System.currentTimeMillis();
        CuestionarioEntity actualizado = new CuestionarioEntity(
                existing.getCuestionarioId(),
                existing.getPacienteId(),
                existing.getPlantillaCodigo(),
                estado,
                respuestas,
                existing.getFirmas(),
                existing.getAdjuntos(),
                existing.getCreatedAt(),
                existing.getUpdatedAt(),
                now,
                existing.getSyncToken(),
                true);
        cuestionarioDao.upsert(actualizado);
        return actualizado;
    }
}

File: android-app/src/main/java/com/prevengos/plug/android/data/repository/CuestionarioRepository.kt
package com.prevengos.plug.android.data.repository

import com.prevengos.plug.android.data.local.dao.CuestionarioDao
import com.prevengos.plug.android.data.local.entity.CuestionarioEntity
import com.prevengos.plug.android.data.local.entity.RespuestaLocal
import kotlinx.coroutines.flow.Flow
import java.util.UUID

class CuestionarioRepository(private val cuestionarioDao: CuestionarioDao) {
    fun observeForPaciente(pacienteId: String): Flow<List<CuestionarioEntity>> =
        cuestionarioDao.observeByPaciente(pacienteId)

    suspend fun createDraft(pacienteId: String, plantillaCodigo: String, respuestas: List<RespuestaLocal>): CuestionarioEntity {
        val now = System.currentTimeMillis()
        val entity = CuestionarioEntity(
            cuestionarioId = UUID.randomUUID().toString(),
            pacienteId = pacienteId,
            plantillaCodigo = plantillaCodigo,
            estado = "borrador",
            respuestas = respuestas,
            firmas = emptyList(),
            adjuntos = emptyList(),
            createdAt = null,
            updatedAt = null,
            lastModified = now,
            syncToken = null,
            isDirty = true
        )
        cuestionarioDao.upsert(entity)
        return entity
    }
}
